### PR TITLE
CDN download

### PR DESCRIPTION
**Pull request by kruplm**
Adding install script that downloads and decompresses the binary.
* skipping the download if the COMPOSEUI_SKIP_DOWNLOAD environment variable is set to true
* CDN URL can be provided via environment variable
* Version is specified by the version number in the package.json or can be overridden by the COMPOSEUI_VERSION variable
* The composeui-node-launcher launches the extracted binary unless an other path is specified by the COMPOSEUI_BINARY_FILE_PATH environment variable.
* The user can specify the variables in an .npmrc file
* README includes documentation and setup for development as well.
* The downloading of the binaries is skipped on the CI since we should provide e2e tests/testing environment later.

### CODE REVIEW

**Review by psmulovics — APPROVED** [review]

File: src/shell/js/composeui-node-launcher/src/cli/install.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+      }
+      console.log(`Extracting zip contents to ${outPath}.`);
+      try {
+        fs.createReadStream(zipFile).pipe(unzipper.Extract({ path: outPath }));
+      } catch (error) {
+        throw new Error('Error extracting archive: ' + error);
+      }
+}
+
+function createTempFolder(){
```
Comment (psmulovics) [inline src/shell/js/composeui-node-launcher/src/cli/install.js:86]: My only concern is, who cleans up the temp folders?

**Review by psmulovics — DISMISSED** [review]

File: src/shell/js/composeui-node-launcher/README.md [hunk 2]
```diff
@@ -48,13 +48,61 @@ The CLI enables you to execute your app with compose by executing the following
...
+
 If you're developing the CLI itself you need to execute the following command
 
 ```
 npm link
 ```
 
-in the `.\Tryouts\Prototypes\Shell\Node-launcher\Lib` folder
\ No newline at end of file
+in the `.\Tryouts\Prototypes\Shell\Node-launcher\Lib` folder
```
Comment (BalassaMarton) [inline src/shell/js/composeui-node-launcher/README.md:90]: The correct path is `src/shell/js/composeui-node-launcher/`

File: src/shell/js/composeui-node-launcher/src/cli/install.js (hunk 1 above)
Comment (BalassaMarton) [reply src/shell/js/composeui-node-launcher/src/cli/install.js:86]: If it's under the os temp directory, isn't it automatically cleaned up after some time?

Comment (psmulovics) [reply src/shell/js/composeui-node-launcher/src/cli/install.js:86]: You run into the 64k files issue soon. E.g. although you hardly run out of space there, windows' builtin functions start to break down when you hit 64k files in that folder.

**Review by psmulovics — APPROVED** [review]

**Pull request merged**